narayana-spring-boot-core/src/main/java/me/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java

```diff
@@ -104,6 +106,15 @@ private void setXAResourceRecordWrappingPlugin(String xaResourceRecordWrappingPl
         getPopulator(JTAEnvironmentBean.class).setXaResourceRecordWrappingPluginClassName(xaResourceRecordWrappingPlugin);
     }
 
+    private void setLastResourceOptimisationInterface(String lastResourceOptimisationInterface) {
+        getPopulator(JTAEnvironmentBean.class).setLastResourceOptimisationInterfaceClassName(lastResourceOptimisationInterface);
+    }
+
+    private void setCommitMarkableResourceJNDINames(List<String> commitMarkableResourceJNDINames) {
+        getPopulator(JTAEnvironmentBean.class).setPerformImmediateCleanupOfCommitMarkableResourceBranches(true);
```

Comment:
Agroal will not support this. See [AG-214](https://issues.redhat.com/projects/AG/issues/AG-214) for further details. I also don't see an advantage on immediate cleanup, the recovery manager can do it in batch which ends up being more efficient.